File: _Src/Freem/Freem.Entities.UseCases.Contracts/RunningRecords/Update/UpdateRunningRecordRequest.cs
using Freem.Entities.Models.Records;
using Freem.Entities.Relations.Collections;

namespace Freem.Entities.UseCases.Contracts.RunningRecords.Update;

public sealed class UpdateRunningRecordRequest
{
    public UpdateField<RecordName?>? Name { get; init; }
    public UpdateField<RecordDescription?>? Description { get; init; }
    
    public UpdateField<RelatedActivitiesCollection>? Activities { get; init; }
    public UpdateField<RelatedTagsCollection>? Tags { get; init; }

    public bool HasChanges()
    {
        return 
            Name is not null ||
            Description is not null ||
            Activities is not null ||
            Tags is not null;
    }
}

File: _Src/Freem/Freem.Web/src/clients/models/records/running/UpdateRunningRecordRequest.ts
import { UpdateField } from "../../UpdateField";

export class UpdateRunningRecordRequest {
  public readonly name?: UpdateField<string>;
  public readonly description?: UpdateField<string>;
  public readonly activities?: UpdateField<string[]>;
  public readonly tags?: UpdateField<string[]>;

  constructor(
    name?: UpdateField<string>,
    description?: UpdateField<string>,
    activities?: UpdateField<string[]>,
    tags?: UpdateField<string[]>
  ) {
    this.name = name;
    this.description = description;
    this.activities = activities;
    this.tags = tags;
  }
}